diff -ru ../libwx.org//src/stc/gen_iface.py ./src/stc/gen_iface.py
--- ../libwx.org//src/stc/gen_iface.py	2012-12-23 20:23:04 +0100
+++ ./src/stc/gen_iface.py	2012-12-23 20:24:21 +0100
@@ -24,7 +24,7 @@
 if len(sys.argv) > 1 and sys.argv[1] == '--wxpython':
     DOCSTR_DEST   = os.path.abspath('../../../wxPython/src/_stc_gendocs.i')
 else:
-    DOCSTR_DEST   = '/dev/null'
+    DOCSTR_DEST   = None
 
 
 # Value prefixes to convert
@@ -816,7 +816,7 @@
     # write out destination files
     open(h_dest, 'w').write(h_text)
     open(cpp_dest, 'w').write(cpp_text)
-    open(docstr_dest, 'w').write(docstrings)
+    if docstr_dest: open(docstr_dest, 'w').write(docstrings)
 
 
